scripts/common_startup.sh

```diff
@@ -65,30 +77,23 @@ for rmfile in $RMFILES; do
     [ -f "$rmfile" ] && rm -f "$rmfile"
 done
 
-# Check client build state.
-if [ $SKIP_CLIENT_BUILD -eq 0 ]; then
-    gitbranch=$(git rev-parse --abbrev-ref HEAD)
-    if [ "$gitbranch" = "dev" ]; then
-        # We're on dev.  This branch (only, currently) doesn't have build
-        # artifacts.  We should probabably swap to a list of releases?
-        # Compare hash.
-        if [ -f static/client_build_hash.txt ]; then
-            githash=$(git rev-parse HEAD)
-            statichash=$(cat static/client_build_hash.txt)
-            if [ "$githash" = "$statichash" ]; then
-                SKIP_CLIENT_BUILD=1
-            fi
-        fi
-    else
-        # Not on dev.  We're not going to bug people about building.
-        SKIP_CLIENT_BUILD=1
-    fi
-    if [ $SKIP_CLIENT_BUILD -eq 0 ]; then
-        echo "The Galaxy client build is out of date.  Please run 'make client' or your choice of client build target (client-*)."
-        echo "If you're sure you'd like to skip this check, you can run galaxy with the --skip-client-build flag, though this is not recommended as the client and server code will potentially be out of sync."
-        echo "See ./client/README.md in the Galaxy repository for more information, including how to get help if you're having trouble."
-        exit 1
-    fi
+# Determine branch (if using git)
+if command -v git >/dev/null && [ -d .git ]; then
+    GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+    case $GIT_BRANCH in
+        release_*|master)
+            # All non-release branches should build the client as necessary
+            SKIP_CLIENT_BUILD=1
+            ;;
+        *)
+            # Ensure nodeenv is installed
+            DEV_WHEELS=1
+            # SKIP_CLIENT_BUILD will default to false, but can be overridden by the command line argument
+            ;;
+    esac
+else
+    GIT_BRANCH=0
+    DEV_WHEELS=1
```

Comment:
Dev wheels will be installed even if we don't need to build the client, is this desirable?

Reply:
Since we don't know which branch the tarball comes from, we have to always build the client the first time.

Reply:
We should consider committing the client_build_hash.txt in release branches, but this is a good solution for now.